sztp-agent/pkg/secureagent/daemon.go

```diff
@@ -56,60 +57,69 @@ func (a *Agent) RunCommandDaemon() error {
 
 func (a *Agent) performBootstrapSequence() error {
 	var err error
-	if a.GetBootstrapURL() == "" {
+	// check if empty
+	if len(a.GetBootstrapURL()) == 1 && a.GetBootstrapURL()[0] == "" {
+		log.Println("lmao")
 		err = a.getBootstrapURL()
 		if err != nil {
 			return err
 		}
 	}
-	err = a.doRequestBootstrapServerOnboardingInfo()
-	if err != nil {
-		return err
-	}
-	err = a.doHandleBootstrapRedirect()
-	if err != nil {
-		return err
-	}
-	err = a.downloadAndValidateImage()
-	if err != nil {
-		return err
-	}
-	err = a.copyConfigurationFile()
-	if err != nil {
-		return err
-	}
-	err = a.launchScriptsConfiguration(PRE)
-	if err != nil {
-		return err
-	}
-	err = a.launchScriptsConfiguration(POST)
-	if err != nil {
-		return err
+	log.Println("Bootstrap URL: ", a.GetBootstrapURL())
+	bootstrapURLs := a.GetBootstrapURL()
+	for _, bootstrapURL := range bootstrapURLs {
+		bootstrapURLCopy := bootstrapURL
+		err = a.doRequestBootstrapServerOnboardingInfo(&bootstrapURLCopy)
+		if err != nil {
+			log.Println("[ERROR] ", err.Error())
+			continue
+		}
+		err = a.doHandleBootstrapRedirect(&bootstrapURLCopy)
+		if err != nil {
+			log.Println("[ERROR] ", err.Error())
+			continue
+		}
+		err = a.downloadAndValidateImage(&bootstrapURLCopy)
+		if err != nil {
+			log.Println("[ERROR] ", err.Error())
+			continue
+		}
+		err = a.copyConfigurationFile(&bootstrapURLCopy)
+		if err != nil {
+			log.Println("[ERROR] ", err.Error())
+			continue
+		}
+		err = a.launchScriptsConfiguration(PRE, &bootstrapURLCopy)
+		if err != nil {
+			log.Println("[ERROR] ", err.Error())
+			continue
+		}
+		err = a.launchScriptsConfiguration(POST, &bootstrapURLCopy)
+		if err != nil {
+			log.Println("[ERROR] ", err.Error())
+			continue
+		}
+		_ = a.doReportProgress(ProgressTypeBootstrapComplete, "Bootstrap Complete", bootstrapURLCopy)
+		return nil
```

Comment:
this shouldn't return here,
should I store the URLs for which it failed and try again for that?
wdyt?

Reply:
no need. you will get those URLs again in the next iteration from network manager...